frontend/src/metabase-lib/v1/metadata/utils/schema.ts

```diff
@@ -1,18 +1,37 @@
-export const getSchemaName = id => {
+import type { DatabaseId, SchemaId, SchemaName } from "metabase-types/api";
+
+export const getSchemaName = (id: string | null | undefined): SchemaId => {
   return parseSchemaId(id)[1];
 };
 
-export const parseSchemaId = id => {
+type ParseSchemaIdResultOptions = {
+  isDatasets?: boolean;
+};
+
+type ParseSchemaIdResult =
+  | [DatabaseId, SchemaId]
+  | [DatabaseId, SchemaId, ParseSchemaIdResultOptions];
+
+export const parseSchemaId = (
+  id: string | null | undefined,
+): ParseSchemaIdResult => {
   const schemaId = String(id || "");
   const [databaseId, schemaName, encodedPayload] = schemaId.split(":");
-  const result = [databaseId, decodeURIComponent(schemaName)];
+  const result: ParseSchemaIdResult = [
+    parseInt(databaseId, 10),
```

Comment:
`DatabaseId` is `number`, but for unknown reason here it was a string.
Luckily it was only ever used to build the API URL.
I changed it to `number` here and updated unit tests accordingly.